### PR TITLE
suit: add missing doxygen documentation to suit

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR is adding the missing doxygen to the `suit.h` file. Some return code are also updated to consistency and some cbor related function were also renamed, still for consistency.

Documentation rewording/improvement suggestions are welcome.

### Testing procedure

- `make static-test` should not return doxygen error related to suit.h
- the suit workflow is working

### CODE REVIEW

File: sys/include/suit/v4/suit.h [hunk 1]
```diff
@@ -34,17 +34,27 @@
...
 
 /**
  * @brief Supported SUIT manifest version
  */
-#define SUIT_MANIFEST_VERSION        4
-#define SUIT_VERSION        1
+#define SUIT_MANIFEST_VERSION           (4)
+
+/**
+ * @brief Current SUIT version
```
Comment (bergzand) [inline sys/include/suit/v4/suit.h]: Maybe reword this to `Current SUIT serialization format version` and add a reference to https://tools.ietf.org/html/draft-moran-suit-manifest-04#section-8.2 here

File: sys/include/suit/v4/suit.h [hunk 2]
```diff
@@ -91,74 +101,191 @@ typedef enum {
...
-    size_t urlbuf_len;
+    cose_key_t *key;                /**< Ptr to the public key for validation */
+    char *urlbuf;                   /**< Buffer containing the manifest url */
+    size_t urlbuf_len;              /**< Length of the manifest url */
 } suit_v4_manifest_t;
 
-#define SUIT_MANIFEST_HAVE_COMPONENTS 0x1
-#define SUIT_MANIFEST_HAVE_IMAGE 0x2
+/**
+ * @brief Mask used to determine if SUIT manifest contains components
```
Comment (bergzand) [inline sys/include/suit/v4/suit.h]: (also with the doc below) I think I would prefer to call these bit flags instead of masks

File: sys/include/suit/v4/suit.h [hunk 3]
```diff
@@ -91,74 +101,191 @@ typedef enum {
...
+ */
+#define SUIT_MANIFEST_HAVE_COMPONENTS   (0x1)
+/**
+ * @brief Mask used to determine if SUIT manifest contains an image
```
Comment (bergzand) [inline sys/include/suit/v4/suit.h]: Same as above

File: sys/include/suit/v4/suit.h [hunk 4]
```diff
@@ -91,74 +101,191 @@ typedef enum {
...
  * @return              SUIT_OK on parseable manifest
  * @return              negative @ref suit_v4_error_t code on error
  */
 int suit_v4_parse(suit_v4_manifest_t *manifest, const uint8_t *buf, size_t len);
 
+/**
+ * @brief Check a manifest policy
+ *
+ *
+ * @param[in]   manifest    manifest context to store information in
```
Comment (bergzand) [inline sys/include/suit/v4/suit.h]: Should this be `manifest context to check the policy for`?

File: sys/include/suit/v4/suit.h [hunk 5]
```diff
@@ -91,74 +101,191 @@ typedef enum {
...
+ *
+ * @return              0 on valid manifest policy
+ * @return              -1 on invalid manifest policy
+ */
 int suit_v4_policy_check(suit_v4_manifest_t *manifest);
 
-int cbor_map_iterate_init(CborValue *map, CborValue *it);
-int cbor_map_iterate(CborValue *map, CborValue *key, CborValue *value);
+/**
+ * @brief Initialize a cbor container for SUIT
```
Comment (bergzand) [inline sys/include/suit/v4/suit.h]: Maybe something like `Initialize a cbor iterator for SUIT cbor map container parsing`

File: sys/include/suit/v4/suit.h [hunk 6]
```diff
@@ -91,74 +101,191 @@ typedef enum {
...
+ * @param[in]   map     the cbor container
+ * @param[in]   it      the cbor iterator
+ *
+ * @return              SUIT_OK when initialization is successful
+ * @return              SUIT_ERR_INVALID_MANIFEST if the manifest is not a cbor container
+ */
+int suit_cbor_map_iterate_init(CborValue *map, CborValue *it);
+
+/**
+ * @brief Iterate on a cbor container
```
Comment (bergzand) [inline sys/include/suit/v4/suit.h]: `Iterate over a cbor map container`?

File: sys/include/suit/v4/suit.h [hunk 7]
```diff
@@ -91,74 +101,191 @@ typedef enum {
...
+ * @param[out]  it          cbor iterator
+ *
+ * @return              0 on success
+ * @return              -1 if bseq is not a cbor string
+ * @return              CborError code on other cbor parser errors
+ */
 int suit_cbor_subparse(CborParser *parser, CborValue *bseq, CborValue *it);
 
+/**
+ * @brief Helper function for writing bytes on flash a given offset
```
Comment (bergzand) [inline sys/include/suit/v4/suit.h]: … flash *to a specified* offset ?

File: sys/include/suit/v4/suit.h [hunk 8]
```diff
@@ -34,17 +34,27 @@
...
+ * @brief   Buffer size used for Cose
+ */
 #ifndef SUIT_COSE_BUF_SIZE
-#define SUIT_COSE_BUF_SIZE      512
+#define SUIT_COSE_BUF_SIZE              (512U)
 #endif
 
-#define SUIT_V4_COMPONENT_MAX 1
+/**
+ * @brief   Maximum components used for SUIT v4
```
Comment (bergzand) [inline sys/include/suit/v4/suit.h]: Maximum *number of* components used for SUIT v4?

**Review by bergzand — COMMENTED** [review]
Quick review from my side, thanks for updating!

File: sys/include/suit/v4/suit.h (hunk 4 above)
Comment (aabadie) [reply sys/include/suit/v4/suit.h]: ah yes, copy-pasted from above...

**Comment by aabadie** [comment]
Thanks for reviewing quickly @bergzand, I updated the PR with your suggestions and directly amended.

File: sys/suit/v4/cbor.c [hunk 9]
```diff
@@ -88,13 +88,13 @@ int suit_cbor_get_int(const CborValue *it, int *out)
 int suit_cbor_get_string(const CborValue *it, const uint8_t **buf, size_t *len)
 {
     if (!(cbor_value_is_text_string(it) || cbor_value_is_byte_string(it) || cbor_value_is_length_known(it))) {
-        return -1;
+        return SUIT_ERR_INVALID_MANIFEST;
```
Comment (kaspar030) [inline sys/suit/v4/cbor.c:91]: while this might be the same value, at this level, there shouldn't be any manifest define in the code. It is just a cbor helper.

File: sys/include/suit/v4/suit.h [hunk 10]
```diff
@@ -91,74 +104,191 @@ typedef enum {
...
+ * @param[in]   buf     bytes to write
+ * @param[in]   len     length of bytes to write
+ * @param[in]   more    whether more data is comming
+ *
+ * @return              0 on success
+ * @return              <0 on error
+ */
 int suit_flashwrite_helper(void *arg, size_t offset, uint8_t *buf, size_t len,
-                    int more);
+                           int more);
```
Comment (kaspar030) [inline sys/include/suit/v4/suit.h:282]: here and below: indent is one off

**Review by kaspar030 — CHANGES_REQUESTED** [review]

File: sys/suit/v4/cbor.c (hunk 9 above)
Comment (aabadie) [reply sys/suit/v4/cbor.c:91]: But this is now consistent with other `suit_cbor_get_<whatever type>` functions.

File: sys/include/suit/v4/suit.h (hunk 10 above)
Comment (aabadie) [reply sys/include/suit/v4/suit.h:282]: The alignment is now correct with the above line.

Comment (kaspar030) [reply sys/include/suit/v4/suit.h:282]: yes, sorry, in github it looks off.

File: sys/suit/v4/cbor.c (hunk 9 above)
Comment (kaspar030) [reply sys/suit/v4/cbor.c:91]: ok

**Review by kaspar030 — APPROVED** [review]

**Comment by aabadie** [comment]
Thanks @kaspar030 ! @bergzand, are you ok to merge ?

**Review by bergzand — APPROVED** [review]
Ack from my side
